src/main/java/com/exasol/releaserobot/usecases/ReportImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.exasol.releaserobot.usecases;
+
+import static com.exasol.releaserobot.usecases.ReleaseRobotConstants.LINE_SEPARATOR;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * An abstract base for a report.
+ */
+public class ReportImpl implements Report {
+    protected final List<Result> results = new LinkedList<>();
```

Comment:
If you want to track both successful and failed results, it's probably better to have 2 collections instead of one.

Can be done in another PR